builtin/mainmenu/settings/dlg_settings.lua

```diff
@@ -622,12 +622,15 @@ local function buttonhandler(this, fields)
 	if fields.show_technical_names ~= nil then
 		local value = core.is_yes(fields.show_technical_names)
 		core.settings:set_bool("show_technical_names", value)
+		write_settings_if_android()
```

Comment:
I would name this `eager_write_settings` or so in case it's extended in the future